Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+
+Files:
+
+* \<Extension>/\<Module>/MRMLDM/vtkMRML\<MyCustomType>DisplayableManager.h
+* \<Extension>/\<Module>/MRMLDM/vtkMRML\<MyCustomType>DisplayableManager.cxx
+
+Key Points:
+
+* Naming convention for class: `vtkSlicer<MyCustomType>DisplayableManager`
+* Inherits from [`vtkMRMLAbstractDisplayableManager`](http://apidocs.slicer.org/master/classvtkMRMLAbstractDisplayableManager.html).
+* Constructing a new node is same as the [data node](#the-data-node).
+* Override [`void OnMRMLSceneNodeAdded(vtkMRMLNode* node)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#ac0102310f18a3880f00e93b8e37210c9) to watch for if a new node of your type is added to the scene. Add an appropriate widget(s) and widget representation(s) for any display nodes.
+* Override [`void OnMRMLSceneNodeRemoved(vtkMRMLNode* node)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#a92b1149d08ec5099a9ec61c0636f37c4) to watch for if a node of your type is removed from the scene. Remove corresponding widget(s) and widget representation(s).
+* Override [`void OnMRMLSceneEndImport()`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#ae826dc18ec31156f0eae3456608eb1c4) to watch for an MRB file that has finished importing.
+* Override [`void OnMRMLSceneEndClose()`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#a5f8a69a47eacec0cd95857beceb8eb34) to clean up when a scene closes.
+* Override [`void ProcessMRMLNodesEvents(vtkObject *caller, unsigned long event, void *callData)`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#a43a759874ce32846c6dffc52e5941744) to watch for changes in the data node that would require the display to change.
+* Override [`void UpdateFromMRML()`](http://apidocs.slicer.org/master/classvtkMRMLAbstractDisplayableManager.html#a0e691194d46e40d73e598ba318324388) and [`void UpdateFromMRMLScene()`](http://apidocs.slicer.org/master/classvtkMRMLAbstractLogic.html#a50caeb3a373ccd53635ae2c08bc2e63e) to bring the displayable manager in line with the MRML Scene.
+
+#### The storage node
+
+A storage node is responsible for reading and writing data nodes to files. A single data node type can have multiple storage node types associated with it for reading/writing different formats. A storage node will be created for both normal save/load operations for a single data node, as well as when you are saving a whole scene to an MRB.
+
+It is common for a data node’s storage node to also write relevant values out of the display node (colors, opacity, etc) at the same time it writes the data.
+
+:::{note}
+
+The storage node is not sufficient in itself to allow the new data node to be saved/loaded from the normal 3D Slicer save/load facilities; the [reader](#the-reader) and [writer](#the-writer) will help with that.
+
+:::
+
+Files:
+
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>StorageNode.h
+* \<Extension>/\<Module>/MRML/vtkMRML\<MyCustomType>StorageNode.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- MRML
                    |-- vtkMRML<MyCustomType>StorageNode.h
                    |-- vtkMRML<MyCustomType>StorageNode.cxx
```

Reply:
Updated to suggested file structure formatting.